### PR TITLE
Introduce package-specific flows

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

**Comment by fridex** [comment]
First these PRs need to get in:
 * https://github.com/fabric8-analytics/fabric8-analytics-jobs/pull/26
 * https://github.com/fabric8-analytics/fabric8-analytics-server/pull/16

Code tested, waiting for a review. I will add missing code comments tomorrow.

**Comment by fridex** [comment]
[test this]

File: cucoslib/dispatcher/flows/monitorUpstreamFlow.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          to: 'monitorUpstreamAnalysisFlow'
+          condition: &shouldScheduleMonitoringCheck
+            name: 'fieldExist'
+            args:
+              key: 'document_id'
+        - from: 'monitorUpstreamAnalysisFlow'
+          to: 'PackageFinalizeTask'
+      failures:
+        - nodes: 'monitorUpstreamAnalysisFlow'
+          fallback: 'PackageFinalizeTaskError'
```
Comment (jpopelka) [inline cucoslib/dispatcher/flows/monitorUpstreamFlow.yml]: The package analysis results seem to not be stored on S3 (`S3PackageData`).
Shouldn't there be an edge to some *PackageResultCollector* here ?

Comment (fridex) [inline cucoslib/dispatcher/flows/monitorUpstreamFlow.yml]: Good point, forgot it. Thanks!

File: cucoslib/storages/s3_package_data.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from .s3_data_base import S3DataBase
+
+
+class S3PackageData(S3DataBase):
+    @staticmethod
+    def _construct_base_file_name(arguments):
+        """Construct location of EPV in the bucket.
+        
+        :param arguments: arguments as passed to the flow
+        :return: str, EPV location in the bucket
```
Comment (fridex) [inline cucoslib/storages/s3_package_data.py]: Thanks, fixed.

**Comment by fridex** [comment]
@jpopelka introduced storing results on S3 in a separate commit. I had to delete old migrations due to head collision, I will squash these commits once you approve changes (I didn't want to squash it now so you have more clear what has changed).

**Comment by fridex** [comment]
Also I will rename flow to something like `package*` as monitoring is probably not the right naming if we introduce git as an ecosystem.

**Comment by fridex** [comment]
rebased & squashed

**Pull request merged**